packages/components-library/tests/unit/CheckboxFilter.spec.ts

```diff
@@ -0,0 +1,122 @@
+import { createLocalVue, mount } from '@vue/test-utils'
+import { BootstrapVue } from 'bootstrap-vue'
+import CheckboxFilter from '@/components/filters/CheckboxFilter.vue'
+
+const localVue = createLocalVue()
+localVue.use(BootstrapVue)
+const stubs = ['font-awesome-icon']
+const options = async () => {
+  return [
+    { value: 'foo', text: 'Foo' },
+    { value: 'bar', text: 'Bar' },
+    { value: 'baz', text: 'Baz' }
+  ]
+}
+
+const buildWrapper = (value?: string[], maxVisibleOptions?: number) => {
+  const wrapper = mount(CheckboxFilter, {
+    localVue,
+    stubs,
+    propsData: {
+      name: 'name',
+      label: 'label',
+      value,
+      options,
+      maxVisibleOptions
+    },
+    listeners: {
+      input: (newVal: any) => { 
+        wrapper.setProps({ value: newVal });
+    }
+  }
+  })
+  return wrapper
+} 
+
+describe('CheckboxFilter.vue', () => {
+  let wrapper:any
```

Comment:
Wat mijn voorkeur zou hebben, in plaats van beforeEach te nesten om zo iedere keer met andere mount opties te komen, is om propsData aan te passen in het begin van de test. Voor die maxVisibleOptions test bijv:
```
await wrapper.setProps({ maxVisibleOptions: 1 })
```

Reply:
@jelmerveen updates test as discussed